src/InterventionBackend.php

```diff
@@ -815,7 +820,7 @@ public function __destruct()
             $this->image->destroy();
         }
         // remove our temp file if it exists
-        if (file_exists($this->getTempPath() ?? '')) {
+        if (str_starts_with(basename($this->getTempPath()), TEMP_FILE_PREFIX) && file_exists($this->getTempPath() ?? '')) {
             unlink($this->getTempPath() ?? '');
```

Comment:
```suggestion
        $tempPath = $this->getTempPath() ?? '';
        if (str_starts_with(basename($tempPath), InterventionBackend::TEMP_FILE_PREFIX) && file_exists($tempPath)) {
            unlink($tempPath);
        }
```
- Prefix const with class it's on (and not use `self` as we have linting rules that prevent that)
- Prevent passing null instead of blank string to basename()
- Use a variable instead of calling the same method several times